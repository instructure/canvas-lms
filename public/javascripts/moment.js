define(['bower/moment/min/moment-with-locales'], function(moment) {
  // Include locale that is not a part of momentjs proper.
  var mi_nz = moment.defineLocale('mi-nz', {
    months: 'Kohi-tāte_Hui-tanguru_Poutū-te-rangi_Paenga-whāwhā_Haratua_Pipiri_Hōngoingoi_Here-turi-kōkā_Mahuru_Whiringa-ā-nuku_Whiringa-ā-rangi_Hakihea'.split('_'),
    monthsShort: 'Kohi_Hui_Pou_Pae_Hara_Pipi_Hōngoi_Here_Mahu_Whi-nu_Whi-ra_Haki'.split('_'),
    weekdays: 'Rātapu_Mane_Tūrei_Wenerei_Tāite_Paraire_Hātarei'.split('_'),
    weekdaysShort: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    weekdaysMin: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    longDateFormat: {
      LT: 'HH:mm',
      LTS: 'HH:mm:ss',
      L: 'DD/MM/YYYY',
      LL: 'D MMMM YYYY',
      LLL: 'D MMMM YYYY [i] HH:mm',
      LLLL: 'dddd, D MMMM YYYY [i] HH:mm'
    },
    calendar: {
      sameDay: '[i teie mahana, i] LT',
      nextDay: '[apopo i] LT',
      nextWeek: 'dddd [i] LT',
      lastDay: '[inanahi i] LT',
      lastWeek: 'dddd [whakamutunga i] LT',
      sameElse: 'L'
    },
    relativeTime: {
      future: 'i roto i %s',
      past: '%s i mua',
      s: 'te hēkona ruarua',
      m: 'he meneti',
      mm: '%d meneti',
      h: 'te haora',
      hh: '%d haora',
      d: 'he ra',
      dd: '%d ra',
      M: 'he marama',
      MM: '%d marama',
      y: 'he tau',
      yy: '%d tau'
    },
    ordinalParse: /\d{1,2}º/,
    ordinal: '%dº'
  });

  return moment;
});
