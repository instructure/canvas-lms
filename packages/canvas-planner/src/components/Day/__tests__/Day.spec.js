/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import moment from 'moment';
import {Day} from '../index';

it('renders the base component with required props', () => {
  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" alwaysRender />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the friendly name in large text when it is today', () => {
  const today = moment();

  const wrapper = shallow(
    <Day timeZone="America/Denver" day={today.format('YYYY-MM-DD')} />
  );
  expect(wrapper.find('Text').first().props().size).toEqual('large');
});

it('renders the friendlyName in medium text when it is not today', () => {
  const yesterday = moment().add(1, 'days');
  const wrapper = shallow(
    <Day timeZone="America/Denver" day={yesterday.format('YYYY-MM-DD')} />
  );
  expect(wrapper.find('Text').first().props().size).toEqual('medium');
});

it('groups itemsForDay based on context type + context id', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      type: 'Course',
      id: 128,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'San Juan',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'Roll for the Galaxy',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }, {
      title: 'Same id, different type',
      context: {
        type: 'Group',
        id: 256,
        inform_students_of_overdue_submissions: false
      }
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  const groupedItems = wrapper.state('groupedItems');
  expect(groupedItems['Course128'].length).toEqual(1);
  expect(groupedItems['Course256'].length).toEqual(2);
  expect(groupedItems['Group256'].length).toEqual(1);
});

it('renders grouping correctly when having itemsForDay', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      type: 'Course',
      id: 128,
      url:"http://www.non_default_url.com",
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'San Juan',
    context: {
      type: 'Course',
      id: 256,
      url:"http://www.non_default_url.com",
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'Roll for the Galaxy',
    context: {
      type: 'Course',
      id: 256,
      url:"http://www.non_default_url.com",
      inform_students_of_overdue_submissions: true
    }
  }, {
      title: 'Same id, different type',
      context: {
        type: 'Group',
        id: 256,
        inform_students_of_overdue_submissions: false
      }
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  expect(wrapper).toMatchSnapshot();
});
it('groups itemsForDay that have no context into the "Notes" category', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      type: 'Course',
      id: 128,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'San Juan',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'Roll for the Galaxy',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'Get work done!'
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} />
  );
  const groupedItems = wrapper.state('groupedItems');
  expect(groupedItems.Notes.length).toEqual(1);
});

it('groups itemsForDay that come in on prop changes', () => {
  const items = [{
    title: 'Black Friday',
    context: {
      type: 'Course',
      id: 128,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'San Juan',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }];

  const wrapper = shallow(
    <Day timeZone="America/Denver" day="2017-04-25" itemsForDay={items} registerAnimatable={() => {}} deregisterAnimatable={() => {}} />
  );
  let groupedItems = wrapper.state('groupedItems');
  expect(Object.keys(groupedItems).length).toEqual(2);

  const newItemsForDay = items.concat([{
    title: 'Roll for the Galaxy',
    context: {
      type: 'Course',
      id: 256,
      inform_students_of_overdue_submissions: true
    }
  }, {
    title: 'Get work done!'
  }]);

  wrapper.setProps({ itemsForDay: newItemsForDay });
  groupedItems = wrapper.state('groupedItems');
  expect(Object.keys(groupedItems).length).toEqual(3);
});

it('renders nothing when there are no items and the date is outside of two weeks', () => {
  const date = moment.tz("Asia/Tokyo").add(15, 'days');
  const wrapper = shallow(
    <Day timeZone="Asia/Tokyo" day={date.format('YYYY-MM-DD')} itemsForDay={[]} />
  );
  expect(wrapper.type()).toBeNull();
});

it('renders when there are no items but within two weeks', () => {
  const date = moment.tz("Asia/Tokyo").add(13, 'days');
  const wrapper = shallow(
    <Day timeZone="Asia/Tokyo" day={date.format('YYYY-MM-DD')} itemsForDay={[]} />
  );
  expect(wrapper.type).not.toBeNull();
});

it('registers itself as animatable', () => {
  const fakeRegister = jest.fn();
  const fakeDeregister = jest.fn();
  const firstItems = [{title: 'asdf', context: {id: 128, inform_students_of_overdue_submissions: true}, id: '1', uniqueId: 'first'}, {title: 'jkl', context: {id: 256, inform_students_of_overdue_submissions: true}, id: '2', uniqueId: 'second'}];
  const secondItems = [{title: 'qwer', context: {id: 128, inform_students_of_overdue_submissions: true}, id: '3', uniqueId: 'third'}, {title: 'uiop', context: {id: 256, inform_students_of_overdue_submissions: true}, id: '4', uniqueId: 'fourth'}];
  const wrapper = mount(
    <Day
      day={'2017-08-11'}
      timeZone="Asia/Tokyo"
      animatableIndex={42}
      itemsForDay={firstItems}
      registerAnimatable={fakeRegister}
      deregisterAnimatable={fakeDeregister}
      updateTodo={() => {}}
    />
  );
  const instance = wrapper.instance();
  expect(fakeRegister).toHaveBeenCalledWith('day', instance, 42, ['first', 'second']);

  wrapper.setProps({itemsForDay: secondItems});
  expect(fakeDeregister).toHaveBeenCalledWith('day', instance, ['first', 'second']);
  expect(fakeRegister).toHaveBeenCalledWith('day', instance, 42, ['third', 'fourth']);

  wrapper.unmount();
  expect(fakeDeregister).toHaveBeenCalledWith('day', instance, ['third', 'fourth']);
});
