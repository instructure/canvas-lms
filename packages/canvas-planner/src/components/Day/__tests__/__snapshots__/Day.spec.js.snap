// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders grouping correctly when having itemsForDay 1`] = `
<div>
  <Heading
    border="none"
    color="inherit"
    ellipsis={false}
    level="h2"
  >
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
      transform="uppercase"
    >
      Tuesday
    </Text>
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
    >
      April 25, 2017
    </Text>
  </Heading>
  <div>
    <Animatable(Grouping)
      animatableIndex={0}
      items={
        Array [
          Object {
            "context": Object {
              "id": 128,
              "inform_students_of_overdue_submissions": true,
              "type": "Course",
              "url": "http://www.non_default_url.com",
            },
            "title": "Black Friday",
          },
        ]
      }
      theme={
        Object {
          "titleColor": null,
        }
      }
      timeZone="America/Denver"
      url="http://www.non_default_url.com"
    />
    <Animatable(Grouping)
      animatableIndex={1}
      items={
        Array [
          Object {
            "context": Object {
              "id": 256,
              "inform_students_of_overdue_submissions": true,
              "type": "Course",
              "url": "http://www.non_default_url.com",
            },
            "title": "San Juan",
          },
          Object {
            "context": Object {
              "id": 256,
              "inform_students_of_overdue_submissions": true,
              "type": "Course",
              "url": "http://www.non_default_url.com",
            },
            "title": "Roll for the Galaxy",
          },
        ]
      }
      theme={
        Object {
          "titleColor": null,
        }
      }
      timeZone="America/Denver"
      url="http://www.non_default_url.com"
    />
    <Animatable(Grouping)
      animatableIndex={2}
      items={
        Array [
          Object {
            "context": Object {
              "id": 256,
              "inform_students_of_overdue_submissions": false,
              "type": "Group",
            },
            "title": "Same id, different type",
          },
        ]
      }
      theme={
        Object {
          "titleColor": null,
        }
      }
      timeZone="America/Denver"
    />
  </div>
</div>
`;

exports[`renders the base component with required props 1`] = `
<div>
  <Heading
    border="bottom"
    color="inherit"
    ellipsis={false}
    level="h2"
  >
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
      transform="uppercase"
    >
      Tuesday
    </Text>
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
    >
      April 25, 2017
    </Text>
  </Heading>
  <div>
    <Container
      display="block"
      margin="small 0 0 0"
      textAlign="center"
    >
      No "To-Do's" for this day yet.
    </Container>
  </div>
</div>
`;
